src/main/java/org/opensearch/ad/rest/RestDeleteAnomalyResultsAction.java

```diff
@@ -70,21 +88,32 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
         DeleteByQueryRequest deleteRequest = new DeleteByQueryRequest(ALL_AD_RESULTS_INDEX_PATTERN)
             .setQuery(searchSourceBuilder.query())
             .setIndicesOptions(IndicesOptions.LENIENT_EXPAND_OPEN_HIDDEN);
-        return channel -> client.execute(DeleteAnomalyResultsAction.INSTANCE, deleteRequest, ActionListener.wrap(r -> {
-            XContentBuilder contentBuilder = r.toXContent(channel.newBuilder().startObject(), ToXContent.EMPTY_PARAMS);
-            contentBuilder.endObject();
-            channel.sendResponse(new BytesRestResponse(RestStatus.OK, contentBuilder));
-        }, e -> {
-            try {
-                channel.sendResponse(new BytesRestResponse(channel, e));
-            } catch (IOException exception) {
-                logger.error("Failed to send back delete anomaly result exception result", exception);
-            }
-        }));
+        CompletableFuture<BulkByScrollResponse> futureResponse = new CompletableFuture<>();
+        sdkRestClient
+            .execute(
+                DeleteAnomalyResultsAction.INSTANCE,
+                deleteRequest,
+                ActionListener
+                    .wrap(deleteResponse -> futureResponse.complete(deleteResponse), ex -> futureResponse.completeExceptionally(ex))
+            );
+        BulkByScrollResponse deleteByQueryResponse = futureResponse
+            .orTimeout(
+                AnomalyDetectorSettings.REQUEST_TIMEOUT.get(extensionsRunner.getEnvironmentSettings()).getMillis(),
+                TimeUnit.MILLISECONDS
+            )
+            .join();
+
+        XContentBuilder contentBuilder = deleteByQueryResponse
+            .toXContent(JsonXContent.contentBuilder().startObject(), ToXContent.EMPTY_PARAMS);
+        contentBuilder.endObject();
+
+        ExtensionRestResponse extensionRestResponse = new ExtensionRestResponse(request, RestStatus.OK, contentBuilder);
```

Comment:
(Not for this PR) 

We could also have an `okNoContent(RestRequest request)` method. :)

Reply:
Something like the below in BaseExtensionRestHandler?

```java
private ExtensionRestResponse okNoContent(RestRequest request) {
     return new ExtensionRestResponse(request, RestStatus.OK);
}

Reply:
Yeah, but it also needs the empty JSON `{}` so the above comment with an `EMPTY_JSON_XCONTENT` Builder would go there.  I'll create a new issue.